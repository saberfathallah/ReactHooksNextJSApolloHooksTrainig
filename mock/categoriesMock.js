export const CATEGORY = {
    id: "5e89c5f78245cd0d75dec50a",
    level: 1,
    name: "Sport",
    children: [
      {
        children: [],
        id: "5e89c6068245cd0d75dec50c",
        name: "foot",
        level: 2,
      },
    ],
  }