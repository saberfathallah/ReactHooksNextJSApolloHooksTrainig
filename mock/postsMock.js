export const POST = {
  categoryId: "5e89c6458245cd0d75dec50f",
  postId: "5e89c6458245cd0d75dec50f",
  comments: [],
  creatorId: "5e89c2fd8245cd0d75dec501",
  description: "corona france post",
  likes: ["5e8a26784c6b1d234b4c1a3f", "5e89c2fd8245cd0d75dec501"],
  id: "5e89c6878245cd0d75dec510",
  userId: {
    id: "5fac3671dd9b1b079249d522",
    email: "sberrr@gmail.com",
    name: "sberrr",
  },
};

// interface Post {
//   userId: User;
//   categoryId: string;
//   postId: string;
//   comments: Comment[];
//   description: string;
//   id: string;
//   likes: string[];
// }

// categoryId: "5e89c6458245cd0d75dec50f"
// comments: (5) [{…}, {…}, {…}, {…}, {…}]
// description: "corona france post"
// id: "5e89c6878245cd0d75dec510"
// likes: (2) ["5e8a26784c6b1d234b4c1a3f", "5e89c2fd8245cd0d75dec501"]
// userId: {name: "Saber Fathallah", id: "5e89c2fd8245cd0d75dec501", __typename: "User"}
