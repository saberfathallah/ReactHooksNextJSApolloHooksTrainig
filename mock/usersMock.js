export const USER_CONNECTED = {
    id: "5fac3671dd9b1b079249d522",
    email: "sberrr@gmail.com",
    name: "sberrr",
  };