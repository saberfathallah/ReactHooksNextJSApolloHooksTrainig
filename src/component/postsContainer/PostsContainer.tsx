import * as React from 'react';
import { useQuery } from '@apollo/react-hooks';

import GET_SEARCH_QUERY from '@graphql/client/queries/getSearchQuery';
import GET_CURRENT_CATEGORY_ID from '@graphql/client/queries/getCurrentCategoryId';
import GET_USER_DETAILS from '@graphql/users/queries/getUserDetails';

import ContentSearch from '../contentSearch';
import AllPosts from '../allPosts';
import PostsCategory from '../postsCategory';

const PostsContainer: React.FC<{}> = () => {
  const { data, loading: loadingUserDetails } = useQuery(GET_USER_DETAILS);
  const { data: { currentCategoryId }, loading } = useQuery(GET_CURRENT_CATEGORY_ID);
  const { data: dataSearchQuery, loading: loadingSearchQuery } = useQuery(GET_SEARCH_QUERY);

  if (loading || loadingUserDetails || loadingSearchQuery) return <p>laoding....</p>;

  const { getUserDetails: { user: userConnected } } = data;

  if (dataSearchQuery && dataSearchQuery.query) {
    return (
      <ContentSearch
        dataSearchQuery={dataSearchQuery}
        userConnected={userConnected}
      />
    );
  }

  return (
    <>
      {
        currentCategoryId === ''
          ? <AllPosts userConnected={userConnected} />
          : <PostsCategory userConnected={userConnected} currentCategoryId={currentCategoryId} />
      }
    </>
  );
};

export default PostsContainer;
