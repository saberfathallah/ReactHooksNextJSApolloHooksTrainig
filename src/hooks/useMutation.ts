import { useMutation as useMutationApollo } from "@apollo/react-hooks";

export const useMutation = (
  mutation: any,
  { variables, optionCache, ...otherOptions }: any
): any => {
  if (optionCache) {
    const [mutationFunction, { data, error, loading }] = useMutationApollo(
      mutation,
      {
        variables,
        update: (cache, { data }) =>
          optionCache.updateCacheFunction(cache, data, optionCache.variables),
          ...otherOptions,
      }
    );
    return [mutationFunction, { data, error, loading }];
  }

  const [mutationFunction, { data, error, loading }] = useMutationApollo(
    mutation,
    {
      variables,
    }
  );

  return [mutationFunction, { data, error, loading }];
};
