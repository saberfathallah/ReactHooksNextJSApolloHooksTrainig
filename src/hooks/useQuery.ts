import { useQuery as useQueryApollo } from "@apollo/react-hooks";

export const useQuery = (query, options={}): any => {
  const { loading, data, ...rest } = useQueryApollo(query, {
    ...options,
  });

  return { data, loading, ...rest };
}

// export default useQuery;

