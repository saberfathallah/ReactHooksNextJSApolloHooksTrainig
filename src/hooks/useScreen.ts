import { useEffect, useState } from "react";
import { isMobile, isMobileWidth } from "@utils/isBrowser";
import { ESKTOP_BREAKPOINT } from "@constants/dimensions";

const useScreen = (isClient) => {
  const [screen, setScreen] = useState({
    screenWidth: ESKTOP_BREAKPOINT,
    isMobileSize: isMobile,
  });

  useEffect(() => {
    if (!isClient) {
      return;
    }

    // first load
    setScreen({
      screenWidth: window.innerWidth,
      isMobileSize: isMobileWidth(window.innerWidth),
    });

    const handleResize = () => {
      if (isClient) {
        setScreen({
          screenWidth: window.innerWidth,
          isMobileSize: isMobileWidth(window.innerWidth),
        });
      }
    };

    window.addEventListener("resize", handleResize);

    return () => window.removeEventListener("resize", handleResize);
  }, [isClient]);

  return screen;
};

export default useScreen;
