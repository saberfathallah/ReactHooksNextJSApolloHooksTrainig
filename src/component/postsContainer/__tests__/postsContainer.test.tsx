import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import PostsContainer from "../PostsContainer";
import { USER_CONNECTED } from "../../../../mock/usersMock";
import { POST } from "../../../../mock/postsMock";
import { useQuery } from "../../../hooks/useQuery";
import { useMutation } from "../../../hooks/useMutation";

jest.mock("../../../hooks/useQuery", () => ({
  useQuery: jest.fn(),
}));

jest.mock("../../../hooks/useMutation", () => ({
  useMutation: jest.fn(),
}));

// @ts-ignore
useQuery.mockImplementation(() => ({
  data: {
    getUserDetails: { user: USER_CONNECTED },
    getPostsByCategoryId: { posts: [POST] },
  },
}));

// @ts-ignore
useMutation.mockImplementation(() => [() => {}, { loading: false }]);

describe("PostsContainer", () => {
  const props = {
    currentCategoryId: "5e89c6028245cd0d75dec50b",
    data: { getUserDetails: { user: USER_CONNECTED } },
    dataSearchQuery: { query: "" },
    loading: false,
    loadingSearchQuery: false,
    loadingUserDetails: false,
  };

  it("should match snapshot", () => {
    const tree = renderer.create(<PostsContainer {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("should display posts by categoryID", () => {
    const wrapper = mount(<PostsContainer {...props} />);
    expect(wrapper.text()).toContain("corona france post");
  });

  it("should display all posts", () => {
    // @ts-ignore
    useQuery.mockImplementation(() => ({
      data: {
        getUserDetails: { user: USER_CONNECTED },
        getAllPosts: { posts: [POST], totalPosts: 1 },
      },
    }));
    props.currentCategoryId = "";

    const wrapper = mount(<PostsContainer {...props} />);
    expect(wrapper.text()).toContain("corona france post");
  });

  it("should match lading snapshot", () => {
    props.loading = true;

    const tree = renderer.create(<PostsContainer {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("should display loading", () => {
    const wrapper = mount(<PostsContainer {...props} />);
    expect(wrapper.text()).toContain("laoding....");
  });

  it("should match snapshot searchcontent", () => {
    props.loading = false;

    props.dataSearchQuery.query = "corona";
    // @ts-ignore
    useQuery.mockImplementation(() => ({
      data: {
        search: { posts: [POST] },
        laoding: false,
      },
    }));
    const tree = renderer.create(<PostsContainer {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("should display searchcontent", () => {
    const wrapper = mount(<PostsContainer {...props} />);
    expect(wrapper.text()).toContain("corona france post");
  });
});
