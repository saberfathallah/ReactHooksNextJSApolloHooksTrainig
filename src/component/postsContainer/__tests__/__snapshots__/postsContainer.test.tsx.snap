// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsContainer should match lading snapshot 1`] = `
<p>
  laoding....
</p>
`;

exports[`PostsContainer should match snapshot 1`] = `
<div>
  <div>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-avatar"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle"
          >
            <img
              alt="Ted talk"
              className="MuiAvatar-img"
              src="https://pbs.twimg.com/profile_images/877631054525472768/Xp5FAPD5_reasonably_small.jpg"
            />
          </div>
        </div>
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
          >
            sberrr
          </span>
          <span
            className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            5 hours ago
          </span>
        </div>
        <div
          className="MuiCardHeader-action"
        >
          <button
            aria-label="settings"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          corona france post
        </p>
        <button
          aria-label="show 11 new notifications"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-heart-100 MuiIconButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <span
              className="MuiBadge-root"
            >
              <svg
                data-file-name="SvgLikeOff"
                style={
                  Object {
                    "width": "28px",
                  }
                }
              />
              <span
                className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle"
              >
                2
              </span>
            </span>
          </span>
        </button>
        <ul
          className="MuiList-root makeStyles-list-118 MuiList-padding"
        />
        <form
          onSubmit={[Function]}
          style={
            Object {
              "display": "contents",
            }
          }
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            style={
              Object {
                "margin": 8,
              }
            }
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
              data-shrink={true}
              htmlFor="standard-full-width"
              id="standard-full-width-label"
            >
              Ajouter un commentaire
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="standard-full-width"
                name="description"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder=""
                required={false}
                type="description"
                value=""
              />
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-129 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Commenter
              <span
                className="MuiButton-endIcon MuiButton-iconSizeSmall"
              >
                <span
                  aria-hidden={true}
                  className="material-icons MuiIcon-root"
                >
                  send
                </span>
              </span>
            </span>
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`PostsContainer should match snapshot 2`] = `
<p>
  laoding....
</p>
`;

exports[`PostsContainer should match snapshot searchcontent 1`] = `
<div>
  <div>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-avatar"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle"
          >
            <img
              alt="Ted talk"
              className="MuiAvatar-img"
              src="https://pbs.twimg.com/profile_images/877631054525472768/Xp5FAPD5_reasonably_small.jpg"
            />
          </div>
        </div>
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
          >
            sberrr
          </span>
          <span
            className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
          >
            5 hours ago
          </span>
        </div>
        <div
          className="MuiCardHeader-action"
        >
          <button
            aria-label="settings"
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="MuiCardContent-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          corona france post
        </p>
        <button
          aria-label="show 11 new notifications"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-heart-100 MuiIconButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <span
              className="MuiBadge-root"
            >
              <svg
                data-file-name="SvgLikeOff"
                style={
                  Object {
                    "width": "28px",
                  }
                }
              />
              <span
                className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle"
              >
                2
              </span>
            </span>
          </span>
        </button>
        <ul
          className="MuiList-root makeStyles-list-118 MuiList-padding"
        />
        <form
          onSubmit={[Function]}
          style={
            Object {
              "display": "contents",
            }
          }
        >
          <div
            className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            style={
              Object {
                "margin": 8,
              }
            }
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
              data-shrink={true}
              htmlFor="standard-full-width"
              id="standard-full-width-label"
            >
              Ajouter un commentaire
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="standard-full-width"
                name="description"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder=""
                required={false}
                type="description"
                value=""
              />
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-129 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Commenter
              <span
                className="MuiButton-endIcon MuiButton-iconSizeSmall"
              >
                <span
                  aria-hidden={true}
                  className="material-icons MuiIcon-root"
                >
                  send
                </span>
              </span>
            </span>
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
